#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("ros_hz", int_t, 0, "Execution rate", 4, 0, 10)
gen.add("detect_range", double_t, 0, "Filtering range for the passthrough filter", 1.0, 0.0, 20.0)
gen.add("leafsize3d", double_t, 0, "Voxel size for VoxelGrid filter", 0.08, 0.05, 0.1)
gen.add("eps", double_t, 0, "EPS parameter for DBSCAN clustering", 0.1, 0.05, 0.2)
gen.add("minPts", int_t, 0, "MinPts parameter for DBSCAN clustering", 10, 2, 10)
gen.add("MeanK", int_t, 0, " number of neighbors to be considered for averaging", 50, 10, 200)
gen.add("StddevMulThresh", double_t, 0, "standard deviation multiplier", 1, 1, 10)
gen.add("MinNeighborsInRadius", int_t, 0, "Min Neighbors In Radius ", 2, 1, 8)
gen.add("RadiusSearch", double_t, 0, "Radius search range", 0.05, 0.01, 0.1)

exit(gen.generate("dynamic_object_detection", "dynamic_object_detection", "pointcloud"))